packages/api/src/Concept/Concept.resolver.ts

```diff
@@ -0,0 +1,50 @@
+import { Resolver, Query, Args, Mutation } from '@nestjs/graphql';
+import { UseGuards } from '@nestjs/common';
+
+import { ConceptService } from './Concept.service';
+import { Concept, ConceptInput, UpdateConceptInput } from './Concept.entity';
+import { GQLAuthGuard } from '../Auth/GQLAuth.guard';
+import { CurrentUser } from '../User/CurrentUser.decorator';
+import { User } from '../User/User.entity';
+
+@Resolver('concept')
+export class ConceptResolver {
+  constructor(private readonly conceptService: ConceptService) { }
+
+  @UseGuards(GQLAuthGuard)
+  @Query(() => [Concept])
+  getConcepts() {
+    return this.conceptService.findConcepts();
+  }
+
+  @UseGuards(GQLAuthGuard)
+  @Query(() => Concept)
+  getConceptByName(@Args('name') name: string) {
+    return this.conceptService.findByName(name);
+  }
+
+  @UseGuards(GQLAuthGuard)
+  @Mutation(() => Concept)
+  createConcept(@Args('concept') concept: ConceptInput) {
+    return this.conceptService.create(concept);
+  }
+
```

Comment:
This should be locked down to the current user. We don't want others to be able to "learn" things for others. We should also find a way to lock this down somehow so it can only be done once you finish a lesson

Reply:
Not sure what you mean here @tristanMatthias . createConcept allows us to create a concept to the DB. Probably through a query or an admin panel to have those panels. We have learnConcept which is restricted to CurrentUser. Did I understand that right or I missed something?

Reply:
Yeah, this comment should apply to `learnConcept` instead. Apologies